File: ProblemCollection/src/kakao_blind_recruitment2023/ValidityPeriodOfPersonalInformationCollection.py
def solution(today, terms, privacies):
    answer = []
    term = dict(map(lambda a: a.split(), terms))
    day = [12 * 28, 28, 1]
    now = sum(day[i] * int(v) for i, v in enumerate(today.split(".")))
    for i, v in enumerate(privacies):
        date, type = v.split()
        if sum(day[i] * int(v) for i, v in enumerate(date.split("."))) + day[1] * int(term[type]) <= now:
            answer.append(i + 1)
    return answer

File: ProblemCollection/src/kakao_blind_recruitment2023/ValidityPeriodOfPersonalInformationCollection.js
function solution(today, terms, privacies) {
    const answer = [];
    const term = {};
    const day = [12 * 28, 28, 1];
    const now = today.split('.').reduce((p, c, i) => p + c * day[i], 0);
    for (let t of terms) {
        let s = t.split(' ');
        term[s[0]] = +s[1];
    }
    for (let idx = 0; idx < privacies.length; idx++) {
        let s = privacies[idx].split(' ');
        if (
            s[0].split('.').reduce((p, c, i) => p + c * day[i], 0) +
                term[s[1]] * day[1] <=
            now
        ) {
            answer.push(idx + 1);
        }
    }
    return answer;
}
